go/vt/vterrors/vterrors.go

```diff
@@ -307,3 +317,28 @@ func Equals(a, b error) bool {
 func Print(err error) string {
 	return fmt.Sprintf("%v: %v\n", Code(err), err.Error())
 }
+
+type SyntaxError struct {
+	Position int
+	Statement string
```

Comment:
Add a nearToken field to match existing error messaging?

Reply:
I put this data in the Message string.